variables.tf

```diff
@@ -158,34 +158,55 @@ variable "fluentd" {
 variable "opensearch" {
   description = "Opensearch configurations"
   type = object({
-    cluster_name          = string
-    manager               = bool
-    seed_hosts            = list(string)
-    bootstrap_security    = bool
-    initial_cluster       = bool
-    tls                   = object({
+    cluster_name       = string
+    manager            = bool
+    seed_hosts         = list(string)
+    initial_manager_nodes = list(string)
+    bootstrap_security = bool
+    initial_cluster    = bool
+
+    tls = object({
       ca_certificate = string
-      server         = object({
+      server = object({
         key         = string
         certificate = string
       })
-      admin_client   = object({
+      admin_client = object({
         key         = string
         certificate = string
       })
     })
-    auth_dn_fields        = object({
+
+    auth_dn_fields = object({
       admin_common_name = string
       node_common_name  = string
       organization      = string
     })
-    verify_domains      = bool 
-    basic_auth_enabled  = bool
+
+    verify_domains     = bool
+    basic_auth_enabled = bool
+
+    audit = optional(object({
+      enabled      = optional(bool, false)
+      storage_type = optional(string, "")  
```

Comment:
This feels to me like cognitive overhead for the caller.

We could make it external when the "external" input is not null and internal otherwise, which I think would make intuitive sense and then the caller wouldn't have to manage this string with the correct value.